zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -468,13 +480,50 @@ pub(crate) fn put_received_transparent_utxo<P: consensus::Parameters>(
             output.mined_height(),
             address,
             receiving_account,
+            true,
         )
     } else {
         // The UTXO was not for any of our transparent addresses.
         Err(SqliteClientError::AddressNotRecognized(*address))
     }
 }
 
+/// Returns the vector of [`TransactionDataRequest`]s that represents the information needed by the
+/// wallet backend in order to be able to present a complete view of wallet history and memo data.
+pub(crate) fn transaction_data_requests<P: consensus::Parameters>(
+    conn: &rusqlite::Connection,
+    params: &P,
+) -> Result<Vec<TransactionDataRequest>, SqliteClientError> {
+    // We cannot construct transaction data requests for the case where we cannot determine the
+    // height at which to begin, so we require that either the target height or mined height be
+    // set. In practice, we should not encounter entries in `transparent_spend_search_queue`
+    // because under ordinary circumstances, it is populated via a call from
+    // `decrypt_and_store_transaction` on ordinary mined transaction data retrieved from the chain.
+    let mut address_request_stmt = conn.prepare_cached(
+        "SELECT ssq.address, IFNULL(t.target_height, t.mined_height + 1)
```

Comment:
Why is this `+ 1`?

Reply:
On reflection, I think this is an error - I was thinking that the output would not be spent in the same block it's created, but obviously that is not necessarily correct for spending transparent outputs.
```suggestion
        "SELECT ssq.address, IFNULL(t.target_height, t.mined_height)
```